File: Python/Medium/921.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        # no need care about the permutation
        # but need care when closing bracket encounter
        # if there is no more opening bracket need add a count to the answer
        # if there is still opening bracket at the end need add to the answer
        countOpen = 0
        answer = 0

        for c in s:
            if c == "(":
                countOpen += 1
            else:
                if countOpen > 0:
                    countOpen -= 1
                else:
                    answer += 1

        return answer + countOpen

File: Java/Medium/921.java
class Solution {
    public int minAddToMakeValid(String s) {
        // just need count brackets no need care about the permutation
        // if encounter close bracket and have open bracket can minus
        // but if run out of open bracket means need add open bracket and add to answer
        // directly
        // if got leftover open bracket add to the answer
        int countOpen = 0;
        int answer = 0;

        for (char c : s.toCharArray()) {
            if (c == '(') {
                countOpen++;
            } else {
                if (countOpen > 0) {
                    countOpen--;
                } else {
                    answer++;
                }
            }
        }

        return countOpen + answer;
    }
}